src/client.rs

```
@@ -99,19 +99,31 @@ impl Web3 {
     }
 
     pub async fn eth_get_transaction_count(&self, address: Address) -> Result<Uint256, Web3Error> {
-        self.jsonrpc_client
-            .request_method(
-                "eth_getTransactionCount",
-                vec![address.to_string(), "latest".to_string()],
-                self.timeout,
-            )
-            .await
+        //check if the node is still syncing
+        match self.eth_syncing().await? {
+            false => {
+                self.jsonrpc_client
+                    .request_method(
+                        "eth_getTransactionCount",
+                        vec![address.to_string(), "latest".to_string()],
+                        self.timeout,
+                    )
+                    .await
+            }
+            true => Err(Web3Error::SyncingNode(
+                "Cannot perform eth_getTransactionCount".to_string(),
+            )),
+        }
     }
 
     /// Get the median gas price over the last 10 blocks. This function does not
     /// simply wrap eth_gasPrice, in post London chains it also requests the base
     /// gas from the previous block and prevents the use of a lower value
     pub async fn eth_gas_price(&self) -> Result<Uint256, Web3Error> {
+        if let Ok(true) = self.eth_syncing().await {
```

I think for consistency this should also return a syncing error.